sources/api/ecs-settings-applier/src/main.rs

```diff
@@ -0,0 +1,68 @@
+#[macro_use]
+extern crate log;
+
+use std::fs;
+use std::io;
+use std::path::{Path, PathBuf};
+use serde::{Serialize};
+
+// FIXME Get from configuration in the future
+const DEFAULT_API_SOCKET: &str = "/run/api.sock";
+
+#[derive(Serialize, Debug)]
+#[serde(rename_all="PascalCase")]
+struct ECSConfig {
+    #[serde(skip_serializing_if = "Option::is_none")]
+    cluster: Option<String>,
+}
+
+fn main() {
```

Comment:
Pulling this into a function that returns a `Result` would let you use the `?` in a handful of places here.

[You can see this in practice in `pluto`](https://github.com/bottlerocket-os/bottlerocket/blob/f1c1d2379c0f8f49b2d20090e5b6f23c38cb6d5d/sources/api/pluto/src/main.rs#L281-L289).
```suggestion
fn run() -> Result<(), ()> {
```

Reply:
Done!